### PR TITLE
Add colors based on Issue / PR state

### PR DESCRIPTION
**Pull request by wd60622**
### Describe what this PR does / why we need it

Using the state, isDraft, stateReason fields from the graphql API to determine the icon and color of the PR or Issue. This should be consistent with the github interface.

This works for `:Octo issue / pr list / search` and `:Octo search`

### Does this pull request fix one issue?

Related to #485

### Describe how you did it

- Add state, isDraft, stateReason to the related graphql queries
- Determine the icon and color based on these fields
- Implement for all list and search commands

### Describe how to verify it

Run any of the affected commands and see that the icons will be colored based on the state of the PR or Issue.

### Special notes for reviews

This only effects telescope picker

### Checklist

- [x] Passing tests and linting standards
- [x] Documentation updates in README.md and doc/octo.txt

### CODE REVIEW

**Comment by wd60622** [comment]
Preview of `:Octo pr search`

preview

**Comment by wd60622** [comment]
I have the colors wrong. The merge should be purple and there should be red if closed. Might need to pull in some more in query

File: lua/octo/pickers/telescope/entry_maker.lua [hunk 1]
```diff
@@ -4,6 +4,23 @@ local utils = require "octo.utils"
...
+local function get_icon_highlight_group(row)
+  local highlight_group
+  if row.isDraft then
+    highlight_group = "OctoGrey"
+  elseif row.state == "OPEN" then
+    highlight_group = "OctoGreen"
+  elseif row.state == "CLOSED" then
+    highlight_group = "OctoRed"
+  elseif row.state == "MERGED" then
+    highlight_group = "OctoRed"
```
Comment (p5quared) [inline lua/octo/pickers/telescope/entry_maker.lua:18]: Haven't looked at Octo src at all so double check the name but merged PRs are purple.

```suggestion
  elseif row.state == "MERGED" then
    highlight_group = "OctoPurple"
```

File: lua/octo/pickers/telescope/provider.lua [hunk 2]
```diff
@@ -181,6 +181,7 @@ function M.issues(opts, develop)
         opts.preview_title = opts.preview_title or ""
         opts.prompt_title = opts.prompt_title or ""
         opts.results_title = opts.results_title or ""
+
```
Comment (p5quared) [inline lua/octo/pickers/telescope/provider.lua:184]: Nit: unnecessary.
```suggestion
```

File: lua/octo/pickers/telescope/entry_maker.lua (hunk 1 above)
Comment (p5quared) [reply lua/octo/pickers/telescope/entry_maker.lua:18]: Oops, didn't see your comments.

**Comment by wd60622** [comment]
Hi @p5quared, Thanks for the review!

I ended up adding richer set of symbols to match the github interface a bit more. Think it looks way better now

`:Octo pr search`
pr-search

`:Octo issue search`
issue-search

File: lua/octo/pickers/telescope/entry_maker.lua [hunk 3]
```diff
@@ -4,6 +4,54 @@ local utils = require "octo.utils"
 
 local M = {}
 
+-- Symbols found with "Telescope symbols"
+local icons = {
```
Comment (GuillaumeLagrange) [inline lua/octo/pickers/telescope/entry_maker.lua:8]: I am not 100% familiar with lua doctyping, and struggle to find information on this, but is it possible to type items in this to be a table with first element string, and second element a string enum representing the colors ?

If so, we could type this array, as well as the return of `get_icon` function

**Review by GuillaumeLagrange — APPROVED** [review]
Olgtm, you can merge if you agree with the comment.

On a side note, I suggest we default to squash and merge to keep the master git history clean, either this or we make the effort of cleaning up the commits before merging which is a bit overkill for a small project like this.

File: lua/octo/pickers/telescope/entry_maker.lua (hunk 3 above)
Comment (wd60622) [reply lua/octo/pickers/telescope/entry_maker.lua:8]: I struggle to find that resourse as well. I will give it a shot

**Pull request merged**